### PR TITLE
Jv rox 18537 switch e2e tests in stackrox to core bpf

### PR DESCRIPTION
**Pull request by JoukoVirtanen**
This PR will likely be closed. It has been split into three parts which can be found at the following links

Set default COLLECTION_METHOD to ebpf. Small PR. Should be quick to review. (Merged)
https://github.com/openshift/release/pull/41702

Add ebpf jobs
https://github.com/openshift/release/pull/41704

Switch most jobs to core_bpf
https://github.com/openshift/release/pull/41738

## Description

The core_bpf collection method was recently released as a technical preview. In the next release it will be GA. In order to do that we need more confidence that it is working. E2e tests will be switched over to using core_bpf as the collection method. Some e2e tests will be kept as ebpf and the nightly CI runs will use both core_bpf and ebpf.

See the original issue in stackrox/collector

https://github.com/stackrox/collector/issues/1231

See the corresponding PR in stackrox/stackrox https://github.com/stackrox/stackrox/pull/7000

See the proposal for this work https://docs.google.com/document/d/1eS6W8yezfG0gf_o99XU7qTZ0iAWiGj54p8Y_uky9nFw/edit

### Setting everything to core_bpf in the main config file

Used commit https://github.com/openshift/release/commit/5e8b953aba9cab3e68b51d070a4aa459bcaf9927 for openshift/release

and commit https://github.com/stackrox/stackrox/pull/7000/commits/cbe4c187ec01d3cde74de23a5929a170eccff226 for stackrox/stackrox

nightly-ibmcloudz-qa-e2e-tests (Failed. Looks like the test themselves were not run.)
nightly-powervs-qa-e2e-tests (Failed. Looks like the test themselves were not run.)
master-nightly-ibmcloudz-qa-e2e-tests (Failed. There is no job script for ibmcloudz-qa-e2e-tests)
master-nightly-powervs-qa-e2e-tests (Failed. Looks like the test themselves were not run.)
aks-qa-e2e-tests (Passed. core_bpf)
aro-qa-e2e-tests (Failed. 3/579 tests failed. core_bpf)
eks-qa-e2e-tests (Passed. core_bpf)
release-4.0-aro-qa-e2e-tests (Failed. 4/579 tests failed. ebpf)
release-4.1-aro-qa-e2e-tests (Failed. 8/580 failed. ebpf)
release-x.y-aro-qa-e2e-tests (Failed. ebpf)

Summary:
Most tests failed, but these are cases were the tests are known to be flaky or the test is known to always fail. The correct collection method was used in all cases.

### All changes

Used commit https://github.com/openshift/release/commit/2b2fad736e4514589785050b940456541d4e08de for openshift/release

and commit https://github.com/stackrox/stackrox/pull/7000/commits/77023d490db939482efbe62ca1d616349c2947d1 where the default collection method for CI was set to core_bpf.

nightly-ibmcloudz-ebpf-qa-e2e-tests (Failed. There is no job script for ibmcloudz-ebpf-qa-e2e-tests) Removed
nightly-ibmcloudz-qa-e2e-tests (Failed. Test not run)
nightly-powervs-ebpf-qa-e2e-tests (Failed. Tests not run) Removed
nightly-powervs-qa-e2e-tests (Passed) 

Will fill out rest.

### CODE REVIEW

File: ci-operator/step-registry/stackrox/e2e/stackrox-e2e-ref.yaml [hunk 1]
```diff
@@ -21,6 +21,8 @@ ref:
     default: ''
   - name: QA_TEST_TARGET
     default: ''
+  - name: COLLECTION_METHOD
+    default: core_bpf
```
Comment (gavin-stackrox) [inline ci-operator/step-registry/stackrox/e2e/stackrox-e2e-ref.yaml:25]: one thing to consider is that the step is shared with existing release CI. Maybe this should default to "ebpf" until all outstanding releases that might not support "core_bpf" reach end of support. The master, release and nightly configurations can explicitly choose the collection method they want.

Comment (JoukoVirtanen) [reply ci-operator/step-registry/stackrox/e2e/stackrox-e2e-ref.yaml:25]: Good point.

File: ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-master-postsubmits.yaml [hunk 2]
```diff
@@ -10,7 +10,7 @@ postsubmits:
       skip_cloning: true
     labels:
       ci.openshift.io/generator: prowgen
-    max_concurrency: 6
+    max_concurrency: 1
```
Comment (JoukoVirtanen) [inline ci-operator/jobs/stackrox/stackrox/stackrox-stackrox-master-postsubmits.yaml:13]: Check concurrency

**Pull request closed without merging**